rs/sns/testing/src/sns.rs

```diff
@@ -88,31 +324,90 @@ pub async fn upgrade_sns_controlled_test_canister(
+    let modified_test_canister_wasm = modify_wasm_bytes(&test_canister_wasm.bytes(), 42);
+
+    // TODO: @rvem: It's impossible to use 'upgrade_sns_controlled_canister::exec' function to upgrade the canister
+    // using the ic-agent backend on the network run by PocketIC because pocket-ic-server currently doesn't support
+    // calls to the management canister ('aaaaa-aa'), hence for now the upgrade is done using a single 'manage_neuron'
+    // call to the governance canister.
+    // let temp_dir = TempDir::new().unwrap();
+    // let new_wasm_path = temp_dir.path().join("new_test_canister.wasm");
+    // {
+    //     let mut new_wasm_file = File::create(&new_wasm_path).unwrap();
+    //     new_wasm_file
+    //         .write_all(&modified_test_canister_wasm)
+    //         .unwrap();
+    //     new_wasm_file.flush().unwrap();
+    // }
+
+    // let icp = Tokens::from_tokens(10).unwrap();
+    // convert_icp_to_cycles(dev_participant_agent, icp).await;
+
+    // let neuron_id = get_caller_neuron(dev_participant_agent, sns.governance)
+    //     .await
+    //     .unwrap();
+    // let candid_arg = (candid::IDLArgs {
+    //     args: vec![candid::IDLValue::try_from_candid_type(&upgrade_arg).unwrap()],
+    // })
+    // .to_string();
+    // let upgrade_args = UpgradeSnsControlledCanisterArgs {
+    //     sns_neuron_id: Some(ParsedSnsNeuron(neuron_id)),
+    //     target_canister_id: canister_id,
+    //     wasm_path: new_wasm_path,
+    //     candid_arg: Some(candid_arg),
+    //     proposal_url: Url::try_from("https://github.com/dfinity/ic").unwrap(),
+    //     summary: "Upgrade Test canister".to_string(),
+    // };
+    // let UpgradeSnsControlledCanisterInfo { proposal_id, .. } =
+    //     upgrade_sns_controlled_canister::exec(upgrade_args, dev_participant_agent)
+    //         .await
+    //         .expect("Failed to upgrade the canister");
+    // let proposal_id = proposal_id.unwrap();
+
+    let neuron_id = get_caller_neuron(dev_participant_agent, sns.governance)
+        .await
+        .unwrap();
+    let _ = propose_and_wait(
+        dev_participant_agent,
+        neuron_id,
+        sns.governance,
+        Proposal {
+            title: "Upgrade SNS controlled canister.".to_string(),
+            summary: "".to_string(),
+            url: "".to_string(),
+            action: Some(Action::UpgradeSnsControlledCanister(
+                UpgradeSnsControlledCanister {
+                    canister_id: Some(canister_id.get()),
+                    new_canister_wasm: modified_test_canister_wasm,
+                    canister_upgrade_arg: Some(Encode!(&upgrade_arg).unwrap()),
+                    mode: Some(CanisterInstallMode::Upgrade as i32),
+                    chunked_canister_wasm: None,
+                },
+            )),
         },
     )
-    .await;
+    .await
+    .unwrap();
+
+    // wait_for_proposal_execution(dev_participant_agent, sns.governance, proposal_id)
```

Comment:
Is this intended to stay?

Reply:
This part belongs to the todo from the line 329, so I think so, but I guess it should be organized slightly better

Reply:
Nit - Could you add a comment that explains that it's part of the TODO above, please?